src/superscalar.cpp

```diff
@@ -572,9 +573,18 @@ namespace randomx {
 			src_ = dst_ = -1;
 			canReuse_ = groupParIsSource_ = false;
 		}
-
 		SuperscalarInstruction(const SuperscalarInstructionInfo* info) : info_(info) {
 		}
+		
+	public:
+		SuperscalarInstruction(const SuperscalarInstruction& obj1) : info_(obj1.info_)
```

Comment:
First of all, see my comment in the conversation - you're fixing a non-existent bug. Second, if you're adding a copy constructor, you must copy all class fields, not just `info_`. This PR is unlikely to be merged, I'm just pointing out errors in implementation.

Reply:
Thanks for your review. I note your comment that the fix should be in a different place, but resolving the issue of copying all class fields.

Reply:
As i see in your code the SuperscalarInstruction copy constructor is called only 1 location in **void generateSuperscalar(SuperscalarProgram& prog, Blake2Generator& gen)** and that call is **SuperscalarInstruction currentInstruction = SuperscalarInstruction::Null;**. So even if i don't copy all fields it will work same as copying all fields in current execution of BCHUnlimited. But keeping coding standards in mind i feel it is better to copy all fields and have made the changes.

Reply:
For your 1st comment, in software engineering we don't define a bug as non existent. A bug is found only after it has been tested in various environments and not just unit test cases designed to work. So BCHUnlimited is clearly a new environment for your software.